examples/sources/src/main/java/com/hortonworks/streamline/examples/sources/FileReaderSpout.java

```diff
@@ -0,0 +1,223 @@
+    private static final String LAST_NAME = "lastName";
+    private transient SpoutOutputCollector spoutOutputCollector;
+    private final String path;
+    private String delimiter;
+    private String outputStream;
+
+    public FileReaderSpout (String path) {
+        this.path = path;
+    }
+
+    public FileReaderSpout withDelimiter (String delimiter) {
+        if (delimiter == null || delimiter.isEmpty()) {
+            String errorString = "Delimiter must not be empty";
+            LOG.error(errorString);
+            throw new RuntimeException(errorString);
+        }
+        this.delimiter = delimiter;
+        return this;
+    }
+
+    public FileReaderSpout withOutputStream (String outputStream) {
+        if (outputStream == null || outputStream.isEmpty()) {
+            String errorString = "Output stream must not be empty";
+            LOG.error(errorString);
+            throw new RuntimeException(errorString);
+        }
+        this.outputStream = outputStream;
+        return this;
+    }
+
+    @Override
+    public void declareOutputFields(OutputFieldsDeclarer outputFieldsDeclarer) {
+        outputFieldsDeclarer.declareStream(outputStream, new Fields(StreamlineEvent.STREAMLINE_EVENT));
+    }
+
+    @Override
+    public void open(Map map, TopologyContext topologyContext, SpoutOutputCollector spoutOutputCollector) {
+        LOG.info("Opening FileReaderSpout");
+        this.spoutOutputCollector = spoutOutputCollector;
+    }
+
+    @Override
+    public void nextTuple() {
+        BufferedReader br = null;
+        try {
+            String line;
+            br = new BufferedReader(Files.newBufferedReader(Paths.get(path), StandardCharsets.UTF_8));
+            while ((line = br.readLine()) != null) {
+                String[] result = line.split(delimiter);
+                if (result.length != 2) {
+                   LOG.error("Format of input file not as expected. Expecting {} separated first name and last name", delimiter);
+                } else {
+                    Map<String, Object> event = new HashMap<>();
+                    event.put(FIRST_NAME, result[0].trim());
+                    event.put(LAST_NAME, result[1].trim());
+                    List<Object> values = new ArrayList<>();
+                    values.add(new MyStreamlineEvent(event, outputStream));
+                    spoutOutputCollector.emit(outputStream, values, UUID.randomUUID());
+                }
+            }
+        } catch (IOException e) {
+            LOG.error("Got exception while reading file at {}", path);
+            throw new RuntimeException(e);
+        } finally {
+            if (br != null) try {
+                br.close();
+            } catch (IOException e) {
+                LOG.error("Got exception while closing buffered reader");
+                throw new RuntimeException(e);
+            }
+        }
+
+    }
+
+    private static class MyStreamlineEvent implements StreamlineEvent {
```

Comment:
Why do we require user to construct a MyStreamlineEvent class?. They should be able to use the default implementation which we also use internally. If its not exposed to user, we can do it as a part of this PR.

Reply:
Same as above